src/lib.rs

```diff
@@ -61,17 +108,104 @@ pub fn check_version(name: &str, current_version: &str) -> Result<(), Box<dyn st
     Ok(())
 }
 
+#[cfg(test)]
+use mockito;
+
+#[cfg(not(test))]
+const CRATES_IO_URL: &str = "https://crates.io";
+
+fn get_base_url() -> String {
+    #[cfg(not(test))]
+    let url = format!("{}/api/v1/crates", CRATES_IO_URL);
+    #[cfg(test)]
+    let url = format!("{}/api/v1/crates", mockito::server_url());
+    url
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
+    use mockito::mock;
     #[test]
     fn test_latest_version() {
```

Comment:
let's add a test like https://github.com/yeoman/update-notifier/blob/master/test/notify.js#L45 for the formatting of the message